client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-plugin-install/index.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-console */
+import { useDispatch } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import wpcom from 'calypso/lib/wp';
+import { ONBOARD_STORE } from '../../../../stores';
+import type { Step } from '../../types';
+
+const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
+
+const SiteMigrationPluginInstall: Step = function BundleTransfer( { navigation } ) {
+	const { submit } = navigation;
+	const { setPendingAction, setProgress } = useDispatch( ONBOARD_STORE );
+	const site = useSite();
+
+	const siteId = site?.ID;
+
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		setPendingAction( async () => {
+			setProgress( 0 );
+
+			let stopPollingPlugins = false;
+			let pollingPluginsRetry = 0;
+			const maxRetry = 10;
+			let installedPlugins = null;
+
+			// Poll until plugins are installed. Once this is done, it's safe to install our plugin.
+			while ( ! stopPollingPlugins ) {
```

Comment:
You can easy implement the pooling logic with react query like this https://www.tecforfun.com/frameworks/how-to-utilize-usequery-for-polling-background-data-fetching/

Reply:
That's handy but it won't work in this case since we're polling inside of a `useEffect`.

Reply:
> That's handy but it won't work in this case since we're polling inside of a useEffect.


React query uses useEffect underhood and rerender the component on every request state is changed, so a code like (just an example)

```typescript
const [arePluginsReady, setArePluginsReady] = useState(false)
const { data, isSuccess } = useQuery(....)
setProgress (0)

useEffect(() =>{
  if( !isSuccess) {
    return null
   }
  setProgress(20)
   setPending( async () => {
        ... maybeInstallPlugin()
        ... maybeActivePlugin()
       setProgress(100)
   }); 
  
}, [isSuccess])

...rest of the code
```

What do you think?